File: python/detect_capital.py
from string import ascii_lowercase, ascii_uppercase


class Solution:
    # using string module
    def detectCapitalUse(self, word: str) -> bool:
        return all(c in ascii_lowercase for c in word) or \
            all(c in ascii_uppercase for c in word) or \
            (word[0] in ascii_uppercase and all(c in ascii_lowercase for c in word[1:]))
    # using methods of str type
    def detectCapitalUse(self, word: str) -> bool:
        return word.islower() or word.isupper() or word.istitle()

File: rust/src/detect_capital.rs
impl Solution {
    pub fn detect_capital_use(word: String) -> bool {
        let title_case = (word.chars().nth(0).unwrap().is_uppercase()) &
            // This slicing is safe since the LC input will be ASCII only
            // In real project code I would probably use an external crate or extend this over
            // multiple lines (maybe by creating a Chars iterator and then advancing it by one)?
            (word[1..].chars().all(|c| c.is_lowercase()));
        let all_uppercase = word.chars().all(|c| c.is_uppercase());
        let all_lowercase = word.chars().all(|c| c.is_lowercase());
        return title_case || all_uppercase || all_lowercase;
    }
}
